gapic-generator/lib/gapic/grpc_service_config/parser.rb

```diff
@@ -0,0 +1,129 @@
+
+          method_configs_json.each do |method_config_json|
+            method_config = parse_config method_config_json
+            service_names = parse_service_names method_config_json[NAMES_JSON_KEY]
+            service_method_names = filter_service_method_names method_config_json[NAMES_JSON_KEY]
+
+            service_names.each do |service_name|
+              service_level_result[service_name] = method_config
+            end
+
+            service_method_names.each do |service_method_name|
+              service_name = service_method_name[SERVICE_NAME_JSON_KEY]
+              method_name = service_method_name[METHOD_NAME_JSON_KEY]
+
+              service_method_level_result[service_name] ||= {}
+              service_method_level_result[service_name][method_name] = method_config
+            end
+          end
+        end
+
+        ServiceConfig.new service_level_result, service_method_level_result
+      end
+
+      private_class_method def self.parse_service_names method_config_json_names
+        service_names_jsons = method_config_json_names.select do |names_json|
+          names_json.size == 1 && names_json.key?(SERVICE_NAME_JSON_KEY)
+        end
+
+        service_names_jsons.map { |names_json| names_json[SERVICE_NAME_JSON_KEY] }
+      end
+
+      private_class_method def self.filter_service_method_names method_config_json_names
+        method_config_json_names.select do |names_json|
+          names_json.size == 2 && names_json.key?(SERVICE_NAME_JSON_KEY) && names_json.key?(METHOD_NAME_JSON_KEY)
+        end
+      end
+
+      private_class_method def self.parse_config method_config_json
+        timeout_seconds = parse_interval_seconds method_config_json[TIMEOUT_JSON_KEY]
+        retry_policy = parse_retry_policy method_config_json[RETRY_POLICY_JSON_KEY]
+
+        MethodConfig.new timeout_seconds, retry_policy
+      end
+
+      private_class_method def self.parse_retry_policy retry_policy_json
+        if retry_policy_json.nil? || retry_policy_json.empty?
+          nil
+        else
+          initial_delay_seconds = parse_interval_seconds retry_policy_json[INITIAL_DELAY_JSON_KEY]
+          max_delay_seconds = parse_interval_seconds retry_policy_json[MAX_DELAY_JSON_KEY]
+          multiplier = retry_policy_json[MULTIPLIER_JSON_KEY]
+          status_codes = retry_policy_json[STATUS_CODES_JSON_KEY]
+
+          RetryPolicy.new initial_delay_seconds, max_delay_seconds, multiplier, status_codes
+        end
+      end
+
+      private_class_method def self.parse_interval_seconds timestring
+        if timestring.empty?
+          nil
+        else
+          timestring_nos = timestring.delete_suffix "s"
+          unless valid_float? timestring_nos
+            error_text = "Was not able to convert the string `#{timestring}` " \
+                         "to a time interval when parsing a grpc service config"
+            raise ParsingError error_text
```

Comment:
Missing comma (`ParsingError` and `error_text` are arguments to the `raise` method.)
```ruby
raise ParsingError, error_text
```